tsdb/series_set.go

```diff
@@ -197,13 +197,33 @@ func (s *SeriesIDSet) UnmarshalBinary(data []byte) error {
 	return s.bitmap.UnmarshalBinary(data)
 }
 
+// UnmarshalBinaryUnsafe unmarshals data into the set.
+// References to the underlying data are used so data should not be reused by caller.
+func (s *SeriesIDSet) UnmarshalBinaryUnsafe(data []byte) error {
+	s.RLock()
+	defer s.RUnlock()
+	return s.bitmap.UnmarshalBinaryUnsafe(data)
+}
+
 // WriteTo writes the set to w.
 func (s *SeriesIDSet) WriteTo(w io.Writer) (int64, error) {
 	s.RLock()
 	defer s.RUnlock()
 	return s.bitmap.WriteTo(w)
 }
 
+// Slice returns a slice of series ids.
+func (s *SeriesIDSet) Slice() []uint64 {
+	s.RLock()
+	defer s.RUnlock()
+
+	var a []uint64
```

Comment:
This can be sized to `s.bitmap.GetCardinality()`.

Reply:
Having said that, I think this is only called from `SeriesIDs()` on the tag value block, which is only called from within tests. 

Given that, I think it would be better to return a `[]uint32` here and then have the caller to the slice allocation and conversion. The reason being that if we expose this API here it may end up getting consumed and it allocates a `[]uint64` each time.

Reply:
The rest of the `SeriesIDSet` API uses `[]uint64` so it seems strange to me to return `[]uint32`.